x-pack/plugins/ingest_manager/server/services/epm/agent/agent.ts

```diff
@@ -5,12 +5,60 @@
  */
 
 import Handlebars from 'handlebars';
+import { safeLoad } from 'js-yaml';
+import { DatasourceConfigRecord } from '../../../../common';
 
-interface StreamVars {
-  [k: string]: string | string[];
+function replaceVariablesInYaml(yamlVariables: { [k: string]: any }, yaml: any) {
+  const yamlKeys = Object.keys(yamlVariables);
+  if (yamlKeys.length === 0) {
+    return yaml;
+  }
+
+  Object.entries(yaml).forEach(([key, value]: [string, any]) => {
+    if (typeof value === 'object') {
+      yaml[key] = replaceVariablesInYaml(yamlVariables, value);
+    }
+
+    if (typeof value === 'string' && yamlKeys.indexOf(value) >= 0) {
+      yaml[key] = yamlVariables[value];
+    }
+  });
+
+  return yaml;
 }
 
-export function createStream(vars: StreamVars, streamTemplate: string) {
-  const template = Handlebars.compile(streamTemplate);
-  return template(vars);
+export function createStream(variables: DatasourceConfigRecord, streamTemplate: string) {
```

Comment:
This function went from returning a YAML string out to returning JS object. I'd like to keep some function that accepts a template string & variables and returns YAML. It doesn't need to this `createStream` function, but I'd like to keep a function which accepts the template string & variables and outputs the YAML. That way we don't lose the YAML output step and can do `safeLoad` if/when we want to convert it to JS.